### ISSUE TITLE
[BUG] Duplicates GitLens Code Lens

### ISSUE THREAD

**Issue by zarifpour** [Ackee-Blockchain/tools-for-solidity-vscode#33]
<img width="984" alt="CleanShot 2022-11-30 at 19 53 37@2x" src="https://user-images.githubusercontent.com/16494335/204939693-5c959938-1532-4f72-b52a-becf63d204bd.png">

Notice: "You, 8 hours ago | You, 8 hours ago..."

**Comment by hacker-DOM** [comment]
Hey @zarifpour thanks for trying out Tools for Solidity! At first sight, this looks like an issue in another extension, are you sure it only happens when TfS is enabled? In any case, we're working on it.

**Comment by zarifpour** [comment]
Hey @hacker-DOM, thanks for the reply. 

I think the extension is great!

I do have a lot of extensions... but as soon as I disable T4S the issue is resolved.

It may be important to note that it is visible on functions and not on the top of Solidity files.

<img width="895" alt="CleanShot 2022-12-01 at 13 48 41@2x" src="https://user-images.githubusercontent.com/16494335/205135520-790bdc86-3595-4384-a867-5dc88a86cc27.png">

<img width="992" alt="CleanShot 2022-12-01 at 13 48 50@2x" src="https://user-images.githubusercontent.com/16494335/205135560-075a0fe8-1375-4b80-9825-0da8a90cc6d4.png">

**Comment by michprev** [comment]
Hello @zarifpour, thank you for using our extension!

I was able to reproduce your issue with the following observations:
- the issue is only reproducible with both Tools for Solidity and Solidity Visual Developer extensions enabled
- I have analyzed debug logs from our extension and have not found anything suspicious
- I have analyzed debug logs from the GitLens extension
    - I can confirm that the GitLens extension is sending code lens twice (it is not a bug in VS Code)
    - noticed the following line in logs `GitCodeLensProvider.provideCodeLenses: — 1 symbol(s) found`

From this, I assume that the GitLens extension uses [Documents symbols](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentSymbol) to find functions and other symbols in source code. Since both Solidity extensions (Tools for Solidity and Solidity Visual Developer) offer Document symbols, the GitLens extension receives duplicated data, resulting in duplicated GitLens code lens. The most reasonable solution seems to be to ask the GitLens developers to try to de-duplicate Document symbols before using them.
